lib/errorResponseInterceptor.ts

```diff
@@ -14,15 +14,26 @@
  * limitations under the License.
  */
 
+const { pipeline } = require('node:stream/promises');
+const { Writable, Readable } = require('node:stream');
+
 export function errorResponseInterceptor(axiosError) {
   if (
     axiosError.response && // must have a response
     axiosError.response.status >= 400 &&
     // must have data:
     axiosError.response.data &&
-    // must be a JSON
+    // must be a JSON request
     // which also implies Content-Type starts with application/json:
-    axiosError.config.responseType === 'json' &&
+    (axiosError.config.responseType === 'json' ||
+      // or a stream request with application/json
+      // that has had the response converted to an object
+      (axiosError.config.responseType === 'stream' &&
+        axiosError.response.headers['content-type'].startsWith(
+          'application/json'
+        ) &&
+        !(axiosError.response.data instanceof Readable) &&
```

Comment:
I think this is redundant with below condition:
```
axiosError.response.data instanceof Object
```

Reply:
`Readable` is also an `Object` so we still need this check.